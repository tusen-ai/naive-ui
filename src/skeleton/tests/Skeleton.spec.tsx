import { mount } from '@vue/test-utils'
import { NSkeleton } from '../index'

describe('n-skeleton', () => {
  it('should work with import on demand', () => {
    mount(NSkeleton)
  })

  it('should work with `text` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        text: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with box', async () => {
    const wrapper = mount(NSkeleton)
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `sharp` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        sharp: false
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `round` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        round: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `circle` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        circle: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `repeat` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        repeat: 2
      }
    })
    expect(wrapper.findAll('.n-skeleton').length).toBe(2)
    wrapper.unmount()
  })

  it('should work with `size` prop', async () => {
    const wrapper = mount(NSkeleton)

    await wrapper.setProps({ size: 'small' })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'medium' })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'large' })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should show correct width', () => {
    const wrapper = mount(NSkeleton, {
      props: {
        circle: false,
        width: 100
      }
    })

    expect(wrapper.find('.n-skeleton').attributes('style')).toContain(
      'width: 100px;'
    )
    wrapper.unmount()
  })

  it('should show correct height', () => {
    const wrapper = mount(NSkeleton, {
      props: {
        circle: false,
        height: 100
      }
    })

    expect(wrapper.find('.n-skeleton').attributes('style')).toContain(
      'height: 100px;'
    )
    wrapper.unmount()
  })

  it('should show correct animation', () => {
    const wrapper = mount(NSkeleton, {
      props: {
        animated: false
      }
    })

    expect(wrapper.find('.n-skeleton').attributes('style')).toContain(
      'animation: none;'
    )
    wrapper.unmount()
  })
})
