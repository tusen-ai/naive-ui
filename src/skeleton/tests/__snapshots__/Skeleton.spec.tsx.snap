// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`n-skeleton > should work with \`circle\` prop 1`] = `"border-radius: 50%; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd;"`;

exports[`n-skeleton > should work with \`round\` prop 1`] = `"border-radius: 4096px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd;"`;

exports[`n-skeleton > should work with \`sharp\` prop 1`] = `"border-radius: 3px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd;"`;

exports[`n-skeleton > should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd; height: 28px;"`;

exports[`n-skeleton > should work with \`size\` prop 2`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd; height: 34px;"`;

exports[`n-skeleton > should work with \`size\` prop 3`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd; height: 40px;"`;

exports[`n-skeleton > should work with \`text\` prop 1`] = `"display: inline-block; vertical-align: -0.125em; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd;"`;

exports[`n-skeleton > should work with box 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-start: #eee; --n-color-end: #ddd;"`;
